apps/studio/src/components/sidebar/ConnectionSidebar.vue

```diff
@@ -217,29 +268,43 @@ const log = rawLog.scope('connection-sidebar');
         return _.orderBy(this.connectionConfigs, this.sortOrder)
       },
       components() {
-        return [
-          this.$refs.savedConnectionList,
-          this.$refs.recentConnectionList
-        ]
+        if (!this.noPins) {
+          return [
+            this.$refs.pinnedConnectionList,
+            this.$refs.savedConnectionList,
+            this.$refs.recentConnectionList
+          ]
+        } else {
+          return [
+            this.$refs.savedConnectionList,
+            this.$refs.recentConnectionList
+          ]
+        }
       }
     },
     mounted() {
-      this.split = Split(this.components, {
-        elementStyle: (dim, size) => ({
-          'flex-basis': `calc(${size}%)`
-        }),
-        direction: 'vertical',
-        sizes: this.sizes
-      })
+      this.split = this.getSplit()
     },
     methods: {
+      getSplit() {
```

Comment:
So sometimes you destroy the existing split before creating a new one and sometimes you don't. I think the reason for that is that you use `getSplit` to simply return a split, but really I think it would be more idiomatic to have it be called `buildSplit`.

I think you can merge it, but here's a minor change that you should make:


```js
methods {
  buildSplit() {
    if (this.split) this.split.destroy()
    this.split = new Split(...)
  },
  ...
}
```
Then you just call it, rather than using it to assign